subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/JavaCompiler.java

```diff
@@ -26,4 +28,17 @@
 @Incubating
 public interface JavaCompiler {
 
+    /**
+     * Returns metadata information about this tool
+     *
+     * @return the tool metadata
+     */
+    JavaToolMetadata getMetadata();
+
+    /**
+     * Returns the path to the executable for this tool
+     *
+     * @return the path to the executable
+     */
+    Path getExecutablePath();
```

Comment:
I still don't quite understand the value of using `Path` here. All other places we deal with use `File`, most Gradle APIs use either plain `File` or some Gradle-specific wrapper (`Directory`, ..) so its hard for me to see the value of using yet another abstraction (whereas I generally agree that `Path` has a nicer API but consumers of this code are usually just passing this value along, not modifying it in any way).

Reply:
I view the `File` use in Gradle APIs as legacy. `Path` should be the norm going forward.

Reply:
Actually, the Gradle idiomatic is to use `RegularFile` and `Directory`. Going to adopt that!